addons/mrp/models/mrp_workorder.py

```diff
@@ -349,6 +355,16 @@ def write(self, values):
                 end_date = fields.Datetime.to_datetime(values.get('date_planned_finished')) or workorder.date_planned_finished
                 if start_date and end_date and start_date > end_date:
                     raise UserError(_('The planned end date of the work order cannot be prior to the planned start date, please correct this to save the work order.'))
+                if 'duration_expected' not in values:
+                    if 'date_planned_start' in values and 'date_planned_finished' in values:
+                        computed_finished_time = self._compute_date_planned_finished(start_date)
```

Comment:
Could you rename the method. It can be confusing with actual compute methods

Reply:
Done ! Changed 'compute' to 'calculate'

Reply:
Hello @Whenrow, are you okay with this PR now ?